### PR TITLE
Reserved words and expanded FormError use

### PR DESCRIPTION
**Pull request by rushsteve1**
**Related Issues**
Never opened an issue for this explicitly but it is related to #5 

**Describe the Changes**
Added a new type of `FormError`: `ReservedWord` and a helper function for it. It checks against a list of words names that are not allowed for projects/users/groups/etc. and raises a `FormError` as needed.

With that I have expanded the use of `FormError` to as many forms as I can so that the `ReservedWord` error and other errors can be presented to the user.

**Still To Do**
Expand error messages to other pages more fully.

### CODE REVIEW

File: src/templates.rs [hunk 1]
```diff
@@ -181,3 +185,13 @@ impl<'v> FromFormValue<'v> for FormError {
         Ok(FormError::from(form_value))
     }
 }
+
+pub const RESERVED_WORDS: &[&str] = &["new", "start", "rcos", "edit"];
+
+pub fn is_reserved(word: &str) -> Result<&str, FormError> {
```
Comment (BrianMejia) [inline src/templates.rs:191]: In this function, add a check to see if the word is just a number? Having a number could cause unintended effects.

**Review by BrianMejia — CHANGES_REQUESTED** [review]
Everything looks to still function, this is a nice catch for stopping unintended redirects. One more case to look out for is to check if the new group/user/event/etc. is just a number.

File: src/templates.rs (hunk 1 above)
Comment (rushsteve1) [reply src/templates.rs:191]: Did this in https://github.com/rcos/observatory-new/pull/98/commits/232a1de540eafd3e49d2fe97a679aa7f453fcbab it was only a one-line change.

**Review by BrianMejia — APPROVED** [review]
Great! Hopefully no more edge cases show up, but in case it does then it shouldn't be too difficult to solve.

**Pull request merged**